linker/symbols.go

```diff
@@ -333,7 +332,8 @@ func sourcePositionForNumber(fd protoreflect.FieldDescriptor) ast.SourcePos {
 	if !ok {
 		return ast.UnknownPos(fd.ParentFile().Path())
 	}
-	numberPath := append(path, internal.FieldNumberTag)
```

Comment:
This pattern is used in quite a few places. https://github.com/go-critic/go-critic/issues/865#issuecomment-522964363 goes through the rationale of this check. I do think we might want to evaluate whether we should create new copied slices at most of these places. I noticed for example this works in sourceinfo due to explicit calls to make/copy in `newLoc*`, however in other places this could mask bugs (since underlying memory could be shared if the slice doesn't need to be recreated.